### DISCUSSION TITLE
PYTHON - df conversion with quote

### DISCUSSION THREAD

**Discussion by reitua7** [DaveSkender/Stock.Indicators#1020] [Help and support]
Hello guys,

I can't convert my dataframe using Quote class:

data= yf.download(XXXX, period="max", start="2010-01-01", end=date.today(), show_errors=False, progress=False)
data["Date"]=data.index
data = data[["Date", "Open", "High", "Low", "Close", "Adj Close", "Volume"]]
data.reset_index(drop=True, inplace=True)
quotes_list = [Quote(d, o, h, l, c, v) for d in zip(data["Date"], data["Open"], data["High"], data["Low"], date["Close"], data["Volume"] )]

i got a " 'NoneType' value cannot be converted", but my dataframe is:
Date         datetime64[ns]
Open                float64
High                float64
Low                 float64
Close               float64
Adj Close           float64
Volume                int64

Do you see my mistake ?

**Comment by DaveSkender** [comment]
Try `float64` on Volume

**Reply by reitua7** [reply]
Thx but it's not working even with float64 on volume.

it's due to this function : CsDecimal.Parse(str(decimal)) but can't solve it...

**Reply by DaveSkender** [reply]
Checkout [the guide](https://python.stockindicators.dev/guide/#using-pandasdataframe).  Try `for d,o,h,l,c,v` instead of `for d`

**Reply by reitua7** [reply]
Yes sorry i failed the copy/paste… it’s what i did but with no sucess

**Reply by DaveSkender** [reply]
I'm sure there's something quite simple but I fail to see it.  @LeeDongGeon1996 any ideas?

**Reply by LeeDongGeon1996** [reply]
@reitua7  Sorry for too late reply. I'm very sure that you have None value in your OHLCV dataset. But our library can accept `None` value as any value of ohlcv. Can you provide us your sample data?

**Reply by LeeDongGeon1996** [reply]
And full error message also will be helpful.

**Comment by DaveSkender** [comment]
> [!TIP]
> For a working example using price quotes from a **pandas.DataFrame**, see our [_Replit_ of the Williams Fractal indicator](https://replit.com/@daveskender/Stock-Indicators-for-Python-Williams-Fractal).  If you're having general trouble getting setup, see our [QuickStart guide](https://github.com/DaveSkender/Stock.Indicators.Python.QuickStart#readme).

See also:

- #1165
